app/src/test/java/com/x8bit/bitwarden/data/autofill/manager/AutofillCompletionManagerTest.kt

```diff
@@ -311,6 +311,98 @@ class AutofillCompletionManagerTest {
         }
     }
 
+    @Suppress("MaxLineLength")
+    @Test
+    fun `completeAutofill when filled partition, organization uses totp, a totp code, and totp generated successfully should build a dataset, place it in a result Intent, copy totp, and finish the Activity`() {
+        val filledData: FilledData = mockk {
+            every { filledPartitions } returns listOf(filledPartition)
+        }
+        val generateTotpResult = GenerateTotpResult.Success(
+            code = TOTP_RESULT_VALUE,
+            periodSeconds = 100,
+        )
+        every { activity.intent } returns mockIntent
+        every { mockIntent.getAutofillAssistStructureOrNull() } returns assistStructure
+        every {
+            autofillParser.parse(
+                autofillAppInfo = autofillAppInfo,
+                assistStructure = assistStructure,
+            )
+        } returns fillableRequest
+        every { cipherView.login?.totp } returns TOTP_CODE
+        every { cipherView.organizationUseTotp } returns true
+        mutableUserStateFlow.value = mockk {
+            every { activeAccount.isPremium } returns false
+        }
+        coEvery {
+            filledDataBuilder.build(autofillRequest = fillableRequest)
+        } returns filledData
+        every {
+            filledPartition.buildDataset(
+                authIntentSender = null,
+                autofillAppInfo = autofillAppInfo,
+            )
+        } returns dataset
+        every { settingsRepository.isAutoCopyTotpDisabled } returns false
+        every { createAutofillSelectionResultIntent(dataset = dataset) } returns resultIntent
+        coEvery {
+            vaultRepository.generateTotp(
+                time = any(),
+                totpCode = TOTP_CODE,
+            )
+        } returns generateTotpResult
+        every {
+            Toast.makeText(
+                context,
+                R.string.verification_code_totp,
+                Toast.LENGTH_LONG,
+            )
+        } returns toast
+
+        autofillCompletionManager.completeAutofill(
+            activity = activity,
+            cipherView = cipherView,
+        )
+
+        verify {
+            activity.setResult(Activity.RESULT_OK, resultIntent)
+            activity.finish()
+        }
+        verify {
```

Comment:
Why split up the verify blocks?

Reply:
cmd c + cmd v from the test above, tldr laziness

Reply:
Ohh, weird